plugins/nf-google/src/main/nextflow/cloud/google/batch/GoogleBatchTaskHandler.groovy

```diff
@@ -510,9 +546,8 @@ class GoogleBatchTaskHandler extends TaskHandler implements FusionAwareTask {
     @Override
     TraceRecord getTraceRecord() {
         def result = super.getTraceRecord()
-        if( jobId && uid ) {
-            result.put('native_id', "$jobId/$uid")
-        }
+        if( jobId && uid )
+            result.put('native_id', "$jobId/$taskId/$uid")
```

Comment:
Wonder if this (and same for AWS batch) can impact Platform somehow

Reply:
Indeed, 

https://github.com/seqeralabs/nf-tower-cloud/blob/0e09eec2de06792c19b41ba3d6460bbe2c4dfc1d/tower-enterprise/src/main/groovy/io/seqera/tower/service/platform/gcp/GoogleLogFetcher.groovy#L72-L75

https://github.com/seqeralabs/nf-tower-cloud/blob/0e09eec2de06792c19b41ba3d6460bbe2c4dfc1d/tower-enterprise/src/main/groovy/io/seqera/tower/service/platform/gcp/GoogleBatchPlatformProvider.groovy#L452-L460


https://github.com/seqeralabs/nf-tower-cloud/blob/0e09eec2de06792c19b41ba3d6460bbe2c4dfc1d/tower-enterprise/src/main/groovy/io/seqera/tower/service/platform/gcp/GoogleBatchJobName.groovy#L30-L32

Reply:
Similar for AWS Batch 

https://github.com/seqeralabs/nf-tower-cloud/blob/0e09eec2de06792c19b41ba3d6460bbe2c4dfc1d/tower-enterprise/src/main/groovy/io/seqera/tower/service/platform/aws/AwsLogFetcher.groovy#L95-L96

Reply:
For aws batch, it should work the same because each child job is just a batch job with its own id.

For google batch, the platform will need to be able to handle a job id (`jobName/uid`) or task id (`jobName/taskId/uid`) based on the number of parts, as there are different APIs for each. It should be possible to support both

Reply:
> For aws batch, it should work the same because each child job is just a batch job with its own id

Do you mean it's the same format as expected by the AWS Batch API? 

 
> For google batch, the platform will need to be able to handle a job id (jobName/uid) or task id (jobName/taskId/uid) based on the number of parts, as there are different APIs for each.

Please open a PR or track it with an issue

Reply:
AWS Batch uses the same APIs for jobs / array jobs / child jobs. Google Batch has different APIs for jobs vs tasks, I will open an issue for the platform.

Reply:
Actually I don't think the google batch provider needs to be updated either. The platform only ever queries the google batch API for the head job, not for tasks. In fact the log stream for tasks is explicitly not supported: https://github.com/seqeralabs/platform/blob/34944581425938e47268b6fcc91778a6ee713d81/tower-enterprise/src/main/groovy/io/seqera/tower/service/platform/ComputePlatformProvider.groovy#L412-L414

Reply:
That's overridden in batch provider https://github.com/seqeralabs/nf-tower-cloud/blob/0e09eec2de06792c19b41ba3d6460bbe2c4dfc1d/tower-enterprise/src/main/groovy/io/seqera/tower/service/platform/gcp/GoogleBatchPlatformProvider.groovy#L450-L462

Reply:
My mistake: https://github.com/seqeralabs/platform/pull/7016

Reply:
Though I wonder, if the logging client is only using the uid to fetch the logs, for a job array it will probably fetch the logs for the entire array instead of a single child task

Reply:
Well no, because the google provider was checking for exactly 2 entries 

```
if( parts.size() != 2 ) throw new IllegalArgumentException("Invalid Google Cloud Batch key '$value'")
```

Your PR fix it. Nice

Reply:
I'm talking about this part: https://github.com/nextflow-io/nextflow/blob/e8b701da5ae0dded9760775158520807dbb0dcc1/plugins/nf-google/src/main/nextflow/cloud/google/batch/logging/BatchLogging.groovy#L69-L83

The platform does a similar thing. It seems to query all log lines with matching job UID, which means for a job array it will fetch the logs for all array tasks.

I think I tried to make it fetch only the i-th task but I couldn't figure out how to modify the query correctly. Couldn't find these fields documented anywhere